### PR TITLE
client selection list and top header changes added

### PR DESCRIPTION
**Pull request by vijayankrishnan**
review code changes and merge if looks good

### CODE REVIEW

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 1]
```diff
@@ -26,13 +26,67 @@
...
+						<div class="col-sm-6">
+						<select class="form-control" id="sel1">
+							<option>Amsterdem Area</option>
+							<option>London</option>
+						</select>
+						</div>
+						<div class="col-sm-6 col-right add-new-customer">
+							<div class="add-new-customer">
+								<span class="glyphicon glyphicon-plus-sign add-icon"></span>
+								<span>Add New Customer</span>
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: "Add New Customer" pull from LOCALE

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 2]
```diff
@@ -26,13 +26,67 @@
...
+						</span>
+					</div>
+				</div>
+			</div>
+			<div class="modal-body">
+			  <div class="customer-list-search">
+					<div class="row form-group">
+						<div class="col-sm-6">
+						<select class="form-control" id="sel1">
+							<option>Amsterdem Area</option>
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: "Amsterdem Area" all visual text/String has to be pulled from LOCALE

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts [hunk 3]
```diff
@@ -8,9 +8,65 @@ import { TranslateService } from 'ng2-translate';
...
 
+	customerList: Array<any>;
+	clientSelectionList = 'inactive';
 	constructor(private translate: TranslateService) {
 	}
 
-	ngOnInit() {}
-
+	ngOnInit() {
+		this.customerList =    [
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts]: try to put this is in External JSON file, later we'll MAP this Data with Service API. Plus make Value Object class instance.

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss [hunk 4]
```diff
@@ -31,3 +33,82 @@ a {
     bottom: 50px;
     width: 100%;
 }
+
+.customer-selection-list.active {
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: just an concern, your CSS covers responsive view !?

Comment (vijayankrishnan) [reply src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: Responsive changes added

**Comment by vijayankrishnan** [comment]
revisit code changes

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 5]
```diff
@@ -26,13 +26,67 @@
...
+	<div  class="customer-selection-list {{clientSelectionList}}">
+		<div id="client-selection-list" class="client_list">
+			<div class="modal-header">
+				<div class="row">
+					<div class="col-sm-10">
+						<i class="fa fa-users fa-2x" aria-hidden="true"></i>
+						<span class="modal-title">clients</span> <span class="modal-title-count">({{customerList.length}})</span>
+					</div>
+					<div class="col-sm-2">
+						 <span class="fa fa-search fa-2x"></span>
```
Comment (kshitizdublish) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: Use icons: http://fontawesome.io/icons/

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss [hunk 6]
```diff
@@ -31,3 +33,82 @@ a {
...
+                font-size: 36px;
+                margin-left: 50px;
+            }
+          }
+          .modal-body {
+            padding: 5px;
+            margin: 15px 0px 50px 0px;
+            .add-new-customer {
+                cursor: pointer;
+                font-size: small;
```
Comment (kshitizdublish) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: We have a variable for it.

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss [hunk 7]
```diff
@@ -31,3 +33,82 @@ a {
...
+                    padding: 10px 5px 10px 10px;
+                    font-size: 14px;
+                }
+            }
+            .customer-list-container {
+            //   .list-group div.row:nth-child(even){
+            //     background-color: #dcdcdc;
+            //   }
+              .list-group div.row:nth-child(odd){
+                background-color:$gray;
```
Comment (kshitizdublish) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: indentation

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss [hunk 8]
```diff
@@ -31,3 +33,82 @@ a {
...
+              }
+              .list-group-item {
+                margin: 0px;
```
Comment (kshitizdublish) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: px not required with 0 value

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts [hunk 9]
```diff
@@ -8,9 +8,65 @@ import { TranslateService } from 'ng2-translate';
...
+			{
+				"site_name":"ABN Bank",
```
Comment (kshitizdublish) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts]: indent

File: src/app/components/dashboard/header/header.component.scss [hunk 10]
```diff
@@ -6,6 +6,7 @@
   background-color: $white;
   margin-bottom: 0;
   height: 55px;
+  box-shadow: 0 3px 4px $gray;
```
Comment (kshitizdublish) [inline src/app/components/dashboard/header/header.component.scss]: I changed $gray to $grey

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 11]
```diff
@@ -26,13 +26,67 @@
...
+		<a (click)="openClientListNav();" title="{{ 'AppLeftPanel.iconTitle.clients' | translate}}">
 			<div aria-hidden="true">
 				<i class="fa fa-users fa-3x" aria-hidden="true"></i>
 			</div>
 		</a>
 	</div>
 
+	<!-- Client List - customer selection menu : starts -->
+	<div  class="customer-selection-list {{clientSelectionList}}">
+		<div id="client-selection-list" class="client_list">
```
[the remark this answers is not in the document]
Comment (vijayankrishnan) [reply src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: changed

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html (hunk 5 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: changed

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss (hunk 6 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: changed

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss (hunk 8 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.scss]: removed

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts (hunk 9 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts]: done already

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 12]
```diff
@@ -32,14 +32,70 @@
...
+					<div class="row form-group">
+						<div class="col-sm-6">
+						<select class="form-control" id="sel1" (change)="onSelectClient($event.target.value)">
+							<option value="0">--Select--</option>
+							<option *ngFor="let cl of customerList" value="{{cl.location}}">{{cl.location}}</option>
+						</select>
+						</div>
+						<div class="col-sm-6 col-right add-new-customer">
+							<div class="add-new-customer">
+								<span class="glyphicon glyphicon-plus-sign add-icon"></span>
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: use icons from fontawesome.io

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 13]
```diff
@@ -32,14 +32,70 @@
...
+								<span>{{ 'ClientsSelectionPanel.addNewCustomer' | translate}}</span>
+							</div>
+						</div>
+					</div>
+			  </div>
+	
+			  <div class="customer-list-container">
+					<div class="list-group">
+						<div class="list-group-item row" *ngFor="let cl of customerList; let i = index ">
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: what do u mean by "cl" here? give some reasonable name like customer

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html [hunk 14]
```diff
@@ -32,14 +32,70 @@
...
+							<div class="col-sm-12 client-col">
+								<h5 class="list-group-item-heading">{{cl.site_name}}</h5>
+								<p class="list-group-item-text">Location: {{cl.location}}</p>
+								<p class="list-group-item-text">Total Sites: {{cl.site_count}}</p>
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.html]: no string has to be hard-coded. Everything has to come for LOCALE. pull Location from LOCALE

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts [hunk 15]
```diff
@@ -1,16 +1,39 @@
 import { Component, OnInit } from '@angular/core';
 import { TranslateService } from 'ng2-translate';
+import { ClientlistService } from '../../../services/clientlist/clientlist.service';
 
 @Component({
 	selector: 'app-left-sidebar',
 	templateUrl: './app-left-sidebar.component.html',
-	styleUrls: ['./app-left-sidebar.component.scss']
+	styleUrls: ['./app-left-sidebar.component.scss'],
+	providers: [ClientlistService]
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts]: don't call any providers on component level. Call on module level.

File: src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts [hunk 16]
```diff
@@ -1,16 +1,39 @@
...
+	ngOnInit() {
+	}
+	
+	openClientListNav() {
+		this.clientSelectionList = 'active';
+		
+		this._clientlistService.getClientList().subscribe( (resp:any) => {
+			//debugger;
+			this.customerList = resp as Array<'client_list'>;
+			//debugger;
```
Comment (happiestRaviThapa) [inline src/app/components/dashboard/app-left-sidebar/app-left-sidebar.component.ts]: don't leave //debugger. anywhere in code.

File: src/app/services/clientlist/clientlist.service.ts [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/observable/throw';
+
+import { CustomEvents } from '../../events/customevents';
+import { AppConfigs } from '../../common/app.config';
+import { Iservices } from '../Iservices.service';
+
+@Injectable()
+export class ClientlistService {
```
Comment (happiestRaviThapa) [inline src/app/services/clientlist/clientlist.service.ts]: implements Iservices{
need to implement  Iservices for every service.

File: src/app/components/dashboard/client-list/client-list.component.ts [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+
+  customerList:  Array<any> = []; 
+	clientSelectionList = 'inactive';
+
+	constructor(private translate: TranslateService, private _customEvents:CustomEvents, private _clientlistService: ClientlistService) {
+	}
+
+	ngOnInit() {
+		this._customEvents.showClientListEvt.subscribe( (value:any)=> {
+			if(value.show == true) {
```
Comment (kshitizdublish) [inline src/app/components/dashboard/client-list/client-list.component.ts]: "if(value.show == true) {" should be "if (value.show == true) {"
Space after 'if'

File: src/app/components/dashboard/client-list/client-list.component.ts [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+				this.openClientListNav()
+			}
+		});
+	}
+	
+
```
Comment (kshitizdublish) [inline src/app/components/dashboard/client-list/client-list.component.ts]: remove 2 extra lines.

File: src/app/components/dashboard/client-list/client-list.component.ts [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+	openClientListNav() {
+		this.clientSelectionList = 'active';
+		
+		this._clientlistService.getClientList().subscribe( (resp:any) => {
+			this.customerList = resp as Array<'client_list'>;		
+		});
+	}
+	// // onSelectClient (value){
+	// // 	this.selectedClientLocation = value;
+	// // }
```
Comment (kshitizdublish) [inline src/app/components/dashboard/client-list/client-list.component.ts]: can you please remove extra //

File: src/app/components/dashboard/client-list/client-list.component.ts (hunk 18 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/client-list/client-list.component.ts]: done

File: src/app/components/dashboard/client-list/client-list.component.ts (hunk 19 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/client-list/client-list.component.ts]: done

File: src/app/components/dashboard/client-list/client-list.component.ts (hunk 20 above)
Comment (vijayankrishnan) [reply src/app/components/dashboard/client-list/client-list.component.ts]: done

**Pull request merged**